recipes/libudev/virtual/conanfile.py

```diff
@@ -0,0 +1,44 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class LibudevConan(ConanFile):
+    name = "libudev"
+    description = "[Virtual package] Installs libudev via a system package manager"
+    topics = ("virtual-package", "libudev")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.freedesktop.org/wiki/Software/systemd/"
+    license = "LGPL-2.1-or-later"
+    
+    settings = "os", "arch"
+
+    def source(self):
+        pass
+
+    def config_options(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("This package only provides support for Linux")
+        if not (tools.os_info.with_apt 
+            or tools.os_info.with_yum
+            or tools.os_info.with_zypper
+            or tools.os_info.with_pacman):
+            raise ConanInvalidConfiguration("Only apt, yum, zypper and pacman are currently supported")
+
+    def system_requirements(self):
+        packages = ""
+        if tools.os_info.with_apt:
+            packages = "libudev-dev"
+        elif tools.os_info.with_yum:
+            packages = "libudev-devel"
+        elif tools.os_info.with_zypper:
+            packages = "libudev-devel"
+        elif tools.os_info.with_pacman:
+            packages = "libsystemd systemd"
+
+        tools.SystemPackageTool().install(packages=packages)
```

Comment:
```suggestion
        tools.SystemPackageTool(conanfile=self).install(packages=packages)
```

Reply:
So SystemPackageTools will check self.settings.arch

Reply:
I was assuming it is doing this already.

Why isn't it default? 🤔

Reply:
Nope. Because the golden rule of keeping same behavior on version 1.x. By default it's None.

https://github.com/conan-io/conan/blob/develop/conans/client/tools/system_pm.py#L14